smaily_for_drupal/src/Form/AdminForm.php

```diff
@@ -154,58 +155,73 @@ public function validateCredentials(array &$form, FormStateInterface $form_state
     }
 
     $subdomain = $this->normalizeSubdomain($subdomain);
-    $ch = curl_init();
-    curl_setopt($ch, CURLOPT_URL,
-      'https://' . $subdomain . '.sendsmaily.net/api/workflows.php?trigger_type=form_submitted'
-    );
-    curl_setopt($ch, CURLOPT_RETURNTRANSFER, TRUE);
-    curl_setopt($ch, CURLOPT_HTTPAUTH, CURLAUTH_BASIC);
-    curl_setopt($ch, CURLOPT_USERPWD, "$username:$password");
-    curl_exec($ch);
-    $http_code = curl_getinfo($ch, CURLINFO_HTTP_CODE);
-    curl_close($ch);
-
-    switch ((int) $http_code) {
-      case 200:
-        // OK code, continue validateCredentials.
-        break;
-
-      case 401:
-        $ajax_response->addCommand(
-          new HtmlCommand(
-            '.smaily_message',
-            '<div class="messages messages--error">' . 'Credentials invalid.' . '</div>')
-        );
-        return $ajax_response;
-
-      case 404:
-        $ajax_response->addCommand(
-          new HtmlCommand(
-            '.smaily_message',
-            '<div class="messages messages--error">' . 'Subdomain error' . '</div>')
-        );
-        return $ajax_response;
-
-      default:
-        $ajax_response->addCommand(
-          new HtmlCommand(
-            '.smaily_message',
-            '<div class="messages messages--error">' . 'Something went wrong.' . '</div>')
-        );
-        return $ajax_response;
+    $url = 'https://' . $subdomain . '.sendsmaily.net/api/workflows.php?trigger_type=form_submitted';
+    
+    try {
+      $client = \Drupal::httpClient();
+      $client->request('GET', $url, [
+        'auth' => [$username, $password],
+      ]);
+
+      $this->config('smaily_for_drupal.settings')
+        ->set('smaily_api_credentials.domain', $subdomain)
+        ->set('smaily_api_credentials.username', $username)
+        ->set('smaily_api_credentials.password', $password)
+        ->save();
+      $ajax_response->addCommand(
+        new HtmlCommand(
+          '.smaily_message',
+          '<div class="messages messages--status">' . 'Credentials valid' . '</div>')
+      );
+      return $ajax_response;
     }
+    catch (ClientException $e) {
+      switch ($e->getResponse()->getStatusCode()) {
```

Comment:
You have a lot of repeating code inside this catch block, let's simplify it:

```php
$errorMessage = 'Something went wrong';

// This can also be replaced with if-elseif statement.
switch ($e->getResponse()->getStatusCode()) {
    case 401:
        $errorMessage = 'Credentials invalid.';
        break;
    case 404:
        $errorMessage = 'Subdomain error';
        break;
}

$ajax_response->addCommand(
    new HtmlCommand(
        '.smaily_message',
        '<div class="messages messages--error">' . $errorMessage . '</div>')
);
return $ajax_response;
```